make/hotspot/gensrc/GensrcLTTng.gmk

```diff
@@ -0,0 +1,17 @@
+LTTNG_SOURCE_DIR := $(TOPDIR)/src/hotspot/share/utilities
+LTTNG_GENSRC_DIR := $(JVM_VARIANT_OUTPUTDIR)/gensrc/lttng
+
+$(LTTNG_GENSRC_DIR)/%.o: $(LTTNG_SOURCE_DIR)/%.cpp
+	$(call MakeDir, $(@D))
+	$(CPP) -I$(LTTNG_SOURCE_DIR) -fpic -c $^ -o $@ 
+
+$(LTTNG_GENSRC_DIR)/lib%.so: $(LTTNG_GENSRC_DIR)/%.o 
+	$(CPP) -shared -o $@ $^ -llttng-ust -ldl
+
+LTTNG_LIB_FILES := $(addprefix $(LTTNG_GENSRC_DIR)/, \
+        liblttng_hs_jni.so \
+        liblttng_hotspot.so \
+        liblttng_hs_private.so \
+    )
+
+TARGETS += $(LTTNG_LIB_FILES)
```

Comment:
JDK uses library make function to define compilation. It's worth use it as it would handle various interesting cases, such as crosscompilation. Example is  make/hotspot/lib/CompileDtraceLibraries.gmk
```
    $(eval $(call SetupNativeCompilation, BUILD_LIBJVM_DTRACE, \
        NAME := jvm_dtrace, \
        OUTPUT_DIR := $(JVM_LIB_OUTPUTDIR), \
        SRC := $(TOPDIR)/src/java.base/solaris/native/libjvm_dtrace, \
        CFLAGS := $(JNI_INCLUDE_FLAGS) -m64 -G -mt -KPIC -xldscope=hidden, \
        LDFLAGS := -m64 -mt -xnolib $(SHARED_LIBRARY_FLAGS), \
        LIBS := $(LIBDL) -lthread -ldoor, \
        OBJECT_DIR := $(LIBJVM_DTRACE_OUTPUTDIR)/objs, \
        DEFINE_THIS_FILE := false, \
    ))
```
Not all parameters are needed for our libs, of course.